File: webapp/API/DTO.ts
export interface collectionDTO {
	collectionId: number;
	userId: number;
	permission: string;
	collectionName: string;
	private: boolean;
}

export interface giffyDTO {
	giffyId: number;
	collectionId: number;
	firebaseUrl: string;
	giffyName: string;
	likes: number;
}

export interface userDTO {
	userId: number;
	userName: string;
	profileImgUrl: string;
	firebaseAuthId: string;
}

File: iOSGiffy/GiffyBoard/DTO.swift
//
//  DTO.swift
//  GiffyBoard
//
//  Created by Hachi on 2023-02-18.
//

import Foundation

struct ErrorDTO: Codable {
    let error: String;
}

struct GiffyDTO: Codable {
    let giffyId: Int
    let collectionId: Int
    let firebaseUrl: String
    let firebaseRef: String
    let giffyName: String
    let likes: Int
}

struct CollectionDTO: Codable, Identifiable {
    let id = UUID()
    let collectionId: Int
    let collectionName: String
    let isPrivate: Bool

    enum CodingKeys: String, CodingKey {
        case collectionId, collectionName, isPrivate = "private"
    }
}

struct GetGiffiesByCollectionIdDTO: Codable {
    let data: [GiffyDTO]
}

struct GetCurrentUserCollectionsDTO: Codable {
    let data: [CollectionDTO]
}
